lib/scripts/configAuditor.js

```diff
@@ -47,78 +44,555 @@ function configAudit (appDir) {
+                    checkMissing(methodsParam, methodsKey, keyStack)
+                    delete methodsParam[methodsKey]
+                    break
+                }
+                keyStack.pop(methodsKey)
+              })
+              // Ignore extra custom log types in methods object
+              ignoreExtras(loggingParam, loggingKey)
+              break
+          }
+          keyStack.pop(loggingKey)
+        })
+        // Ignore extra objects/params in logging object
+        ignoreExtras(userConfig, key)
+        break
+      case 'minify':
+        checkMissing(userConfig, key, keyStack)
+        delete userConfig[key]
+        break
+      case 'htmlValidator':
+        const validatorParam = userConfig[key] || {}
+        const defaultValidatorParam = defaultConfig[key]
+        const validatorKeys = Object.keys(defaultValidatorParam)
+        validatorKeys.forEach((validatorKey) => {
+          keyStack.push(validatorKey)
+          switch (validatorKey) {
+            case 'enable':
+              checkMissing(validatorParam, validatorKey, keyStack)
+              delete validatorParam[validatorKey]
+              break
+            case 'separateProcess':
+              const sepProcParam = validatorParam[validatorKey] || {}
+              const defaultSepProcParam = defaultValidatorParam[validatorKey]
+              const sepProcKeys = Object.keys(defaultSepProcParam)
+              sepProcKeys.forEach((sepProcKey) => {
+                keyStack.push(sepProcKey)
+                switch (sepProcKey) {
+                  case 'enable':
+                    checkMissing(sepProcParam, sepProcKey, keyStack)
+                    delete sepProcParam[sepProcKey]
+                    break
+                  case 'autoKiller':
+                    checkMissing(sepProcParam, sepProcKey, keyStack)
+                    delete sepProcParam[sepProcKey]
+                    break
+                  case 'autoKillerTimeout':
+                    checkMissing(sepProcParam, sepProcKey, keyStack)
+                    delete sepProcParam[sepProcKey]
+                    break
+                }
+                keyStack.pop(sepProcKey)
+              })
+              break
+            case 'port':
+              checkMissing(validatorParam, validatorKey, keyStack)
+              delete validatorParam[validatorKey]
+              break
+            case 'showWarnings':
+              delete validatorParam[validatorKey]
+              break
+            case 'exceptions':
+              const exceptionsParam = validatorParam[validatorKey] || {}
+              const defaultExceptionsParam = defaultValidatorParam[validatorKey]
+              const exceptionsKeys = Object.keys(defaultExceptionsParam)
+              exceptionsKeys.forEach((exceptionKey) => {
+                keyStack.push(exceptionKey)
+                switch (exceptionKey) {
+                  case 'requestHeader':
+                    delete exceptionsParam[exceptionKey]
```

Comment:
is it correct that a case like this where you just delete the key means that the key is optional?

Reply:
The purpose of delete on a param is to prevent it from being considered an extra param. This is because whatever params are left in the `userConfig` are considered extra params

Reply:
This creates the following regression:

1. Create a fresh sample app
2. Set the roosevelt dependency to your branch
3. `npm i`
4. `npm run dev`

App will crash because you deleted a bunch of keys from the config...

You can fix this by copying the object instead of passing by reference.

Reply:
Updated